numba/tests/doc_examples/test_examples.py

```diff
@@ -197,6 +197,19 @@ def func_mt(*args):
             timefunc(correct, "numba (%d threads)" % nthreads, func_nb_mt, a, b)
             # magictoken.ex_no_gil.end
 
+    def test_guvectorize_scalar_return(self):
+        with captured_stdout():
+            # magictoken.ex_guvectorize_scalar_return.begin
+            from numba import guvectorize, int64
+
+            @guvectorize([(int64[:], int64, int64[:])], '(n),()->()')
+            def g(x, y, res):
+                acc = 0
+                for i in range(x.shape[0]):
+                    acc += x[i] + y
+                res[0] = acc
+            # magictoken.ex_guvectorize_scalar_return.end
+
```

Comment:
In addition to checking that the example code executes without error, we can also ensure properties of the example hold - in this case, we can check that the result is an integer and that its value is as expected. This also helps to ensure that the example does what we state it does in the documentation.

For this particular example, we can convert the invocation of `g()` to be part of the doctest, and create a second block enclosed by magic tokens for inclusion into the example documentation - then we can assert that the correct result is given:

```suggestion

            # magictoken.ex_guvectorize_scalar_return_call.begin
            a = np.arange(5)
            result = g(a, 2)
            # At this point, result == 20.
            # magictoken.ex_guvectorize_scalar_return.end

            self.assertIsInstance(result, np.integer)
            self.assertEqual(result, 20)


```